internal/state/resolver/resolver.go

```diff
@@ -58,41 +59,42 @@ func (e *ServiceResolverImpl) Resolve(ctx context.Context, svc *v1.Service, svcP
 		return nil, fmt.Errorf("no endpoints found for Service %s", client.ObjectKeyFromObject(svc))
 	}
 
-	return resolveEndpoints(svc, svcPort, endpointSliceList)
+	return resolveEndpoints(svc, port, endpointSliceList)
 }
 
 func resolveEndpoints(
 	svc *v1.Service,
-	svcPort int32,
+	port int32,
 	endpointSliceList discoveryV1.EndpointSliceList,
 ) ([]Endpoint, error) {
-	targetPort, err := getTargetPort(svc, svcPort)
+	svcPort, err := getServicePort(svc, port)
 	if err != nil {
 		return nil, err
 	}
 
-	capacity := calculateEndpointSliceCapacity(endpointSliceList.Items, targetPort)
+	filteredSlices := filterEndpointSliceList(endpointSliceList, svcPort)
```

Comment:
To calculate the capacity, we filter the list, which allocates `filtered := make([]discoveryV1.EndpointSlice, 0, len(endpointSliceList.Items))`. 

Considering that, would it be easier (and same in terms of memory)  if we just `endpoints := make([]Endpoint, 0, len(endpointSliceList.Items))`

Reply:
There can be multiple endpoints in an EndpointSlice. `endpointSliceList.Items` is just the list of EndpointSlices.

Reply:
my bad. misread the code :(